spec/models/match_spec.rb

```diff
@@ -0,0 +1,77 @@
+require "rails_helper"
+
+RSpec.describe Match, type: :model do
+  let(:teams) { create_list(:team_with_members, 2) }
+  let(:team_a) { teams[0] }
+  let(:team_b) { teams[1] }
+
+  describe "score" do
+    let(:match) { build(:match, team_one_score: 3) }
+    it "is present" do
+      expect(match.team_one_score).to eq(3)
+    end
+
+    context "ongoing match" do
+      let(:match) { build(:ongoing_match) }
+      it "is absent" do
+        expect(match.team_one_score).to be_nil
+      end
+    end
+  end
+
+  describe ".played_at" do
+    let(:match) { build(:match) }
+    it "is a time" do
+      expect(match.played_at).to be_a(Time)
+    end
+  end
+
+  context "with valid data" do
+    context ".no_repeated_members_across_teams" do
+      let(:match) { build(:match, team_one: team_a, team_two: team_b) }
+      it "can have teams with distinct players" do
+        expect(match).to be_valid
+      end
+    end
+  end
+
+  describe "validations" do
+    context ".played_at" do
+      let(:match) { build(:match, played_at: nil) }
+      it "is not missing" do
```

Comment:
this message doesn't give me a feedback as a dev. it says it is not missing but has played_at nil so what is not missing